syntax = "proto3";
package auth;

service Auth{
    rpc Register(RegisterReq) returns (AuthResp);
    rpc Login(LoginReq) returns (AuthResp);
    rpc Validate(ValidateReq) returns (ValidateResp);
}

message RegisterReq{
    string username = 1;
    string password = 2;
}

message LoginReq{
    string username = 1;
    string passwrod = 2;
}

message ValidateReq{
    // empty
}

message AuthResp{
    string token = 1;
    string error = 2;
}

message ValidateResp{
    bool valid = 1;
    string username = 2;
    string error = 3;
}
